native-engine/datafusion-ext-plans/src/common/row_null_checker.rs

```
@@ -0,0 +1,847 @@
+            DataType::UInt64 => FieldConfig::new_primitive(sort_options, 9),
+            DataType::Float32 => FieldConfig::new_primitive(sort_options, 5), // 1 byte null
+            // flag + 4 bytes
+            // value
+            DataType::Float64 => FieldConfig::new_primitive(sort_options, 9), // 1 byte null
+            // flag + 8 bytes
+            // value
+            DataType::Utf8 | DataType::LargeUtf8 => FieldConfig::new_variable(sort_options),
+            DataType::Binary | DataType::LargeBinary => FieldConfig::new_variable(sort_options),
+            DataType::FixedSizeBinary(size) => {
+                FieldConfig::new_fixed_size_binary(sort_options, *size as usize)
+            }
+            DataType::List(_) | DataType::LargeList(_) => FieldConfig::new_list(sort_options),
+            DataType::Struct(_) => {
+                // For struct types, we need to calculate the total encoded length
+                // This is a simplified implementation
+                let estimated_length = 32; // Placeholder for struct encoding length
+                FieldConfig::new_struct(sort_options, estimated_length)
+            }
+            DataType::Dictionary(_, value_type) => {
+                // Dictionary encoding typically uses the key type for encoding
+                // Simplified to use a fixed length based on common key types
+                let key_length = match value_type.as_ref() {
+                    DataType::Int8 | DataType::UInt8 => 2,
+                    DataType::Int16 | DataType::UInt16 => 3,
+                    DataType::Int32 | DataType::UInt32 => 5,
+                    DataType::Int64 | DataType::UInt64 => 9,
+                    _ => 5, // Default to 32-bit key
+                };
+                FieldConfig::new_dictionary(sort_options, key_length)
+            }
+            _ => {
+                // For unsupported types, treat as variable length
+                FieldConfig::new_variable(sort_options)
```

should panic when meeting unsupported types